dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp

```diff
@@ -430,14 +451,14 @@ namespace DYN {
   ModelLoadRestorativeWithLimits::getIndexesOfVariablesUsedForCalculatedVarI(unsigned numCalculatedVar, vector<int>& numVars) const {
     switch (numCalculatedVar) {
       case PNum_:
-        if (isConnected()) {
+        if (static_cast<int>(zLocal_[running]) == RUNNING_TRUE) {
           numVars.push_back(UrYNum_);
           numVars.push_back(UiYNum_);
           numVars.push_back(UfYNum_);
         }
         break;
       case QNum_:
-        if (isConnected()) {
+        if (static_cast<int>(zLocal_[running]) == RUNNING_TRUE) {
```

Comment:
maybe we could keep isConnected() and replace its content with this static cast?